nucliadb/nucliadb/ingest/consumer/pull_v2.py

```diff
@@ -0,0 +1,191 @@
+                await self._loop()
+            except ReallyStopPulling:
+                logger.info("Exiting...")
+                break
+            except Exception as e:
+                errors.capture_exception(e)
+                logger.exception("Exception on worker", exc_info=e)
+                await asyncio.sleep(10)
+
+    async def _loop(self):
+        headers = {}
+        data = None
+        if nuclia_settings.nuclia_service_account is not None:
+            headers[
+                "X-Nuclia-NUAKEY"
+            ] = f"Bearer {nuclia_settings.nuclia_service_account}"
+
+        cursor = None
+        async with self.driver.transaction() as txn:
+            val = await txn.get(DB_TXN_KEY)
+            if val is not None:
+                cursor = val.decode()
+
+        sleep_backoff = 0.1
+        async with ProcessingV2HTTPClient() as processing_http_client:
+            logger.info(f"Collecting from NucliaDB Cloud")
+            while True:
+                try:
+                    await asyncio.sleep(sleep_backoff)
+                    data = await processing_http_client.pull(cursor=cursor, limit=3)
+
+                    if len(data.results) > 0:
+                        for result in data.results:
+                            if result.payload is not None:
+                                await self.handle_message(result.payload)
+                            else:
+                                logger.warning(
+                                    "Payload has expired, skipping.",
+                                    extra={
+                                        "resource_id": result.resource_id,
+                                        "kbid": result.kbid,
+                                        "processing_id": result.processing_id,
+                                    },
+                                )
+                        cursor = data.cursor
+                        async with self.driver.transaction() as txn:
+                            await txn.set(DB_TXN_KEY, cursor.encode())
+                        sleep_backoff = 0.1
+                    else:
+                        sleep_backoff = self.pull_time_empty_backoff
+                except (
+                    asyncio.exceptions.CancelledError,
+                    RuntimeError,
+                    KeyboardInterrupt,
+                    SystemExit,
+                ):
+                    logger.info(f"Pull task for was canceled, exiting")
+                    raise ReallyStopPulling()
+                except ClientConnectorError:
+                    logger.error(
+                        f"Could not connect to processing engine, \
+                         {processing_http_client.base_url} verify your internet connection"
+                    )
+                    sleep_backoff = self.pull_time_error_backoff
+                except Exception:
+                    logger.exception("Unhandled error pulling messages from processing")
```

Comment:
WDYT about adding `errors.capture_exception()` here so that we get a sentry alert?

Reply:
yup, sure